File: P5/Napakalaki/src/napakalaki/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

/**
 *
 * @author Victor Vallecillo Morilla
 */
public class Monster implements Card
{
    private String name;
    private int combatLevel;
    private BadConsequence badConsequence; //Segun el diagrama sería "badConsequence"
    private Prize prize; //Segun el diagrama sería "prize"
    private int levelChangeAgainstCultistPlayer;
    
    private String icon;
    
    public Monster(String name, int combatLevel, String icon) {
        this.name = name;
        this.combatLevel = combatLevel;
        this.icon = icon;
    }
    
    public Monster(String n, int l, BadConsequence b, Prize p, String icon)
    {
        this.name=n;
        this.combatLevel=l;
        this.badConsequence=b;
        this.prize=p;
        this.icon=icon;
    }
    
    public Monster(String n, int l, BadConsequence badConsequence, Prize p, int IC, String icon){ 
        this.name=n;
        this.combatLevel=l;
        this.badConsequence=badConsequence;
        this.prize=p;
        this.levelChangeAgainstCultistPlayer=IC;
        this.icon=icon;
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return this.levelChangeAgainstCultistPlayer + this.getCombatLevel();
    }
    
    public String getName()
    { 
        return  name; 
    }
    
    public int getCombatLevel()
    {
        return combatLevel; 
    }
    
    public BadConsequence getBadConsequence() {
        return badConsequence;
    }
    
    public Prize getPrize() {
        return prize;
    } 
    
    public int getLevelsGained()
    {
        return prize.getLevels();
    }
    
   public int getTreasuresGained()
   {
       return prize.getTreasures();
   }
      
   @Override
   public String toString(){
        return "Name = " + this.name + " CombatLevel = " + Integer.toString(this.combatLevel);
    } 
   
    @Override
    public int getBasicValue() {
        return this.getCombatLevel();
    }

    @Override
    public int getSpecialValue() {
        return this.getCombatLevel() + this.levelChangeAgainstCultistPlayer;
    }

    @Override
    public String getIcon() {
        return "/cartas/monsters/" + this.icon +  ".jpg";
    }
}

File: P4/NapakalakiRuby/lib/monster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'prize'
require_relative 'bad_consequence'
require_relative 'player'
require_relative 'napakalaki_ruby'
require_relative 'dice'
require_relative 'treasure'

module NapakalakiRuby

  class Monster
    attr_reader :name, :combatLevel, :badConsequence, :prize, :levelChangeAgainstCultistPlayer, :ic
    public
      def initialize(name , combatLevel, badConsequence, prize, ic=0)
        @name=name
        @combatLevel=combatLevel
        @badConsequence=badConsequence
        @prize=prize
        @levelChangeAgainstCultistPlayer=ic
      end
      
    
     def getLevelsGained
       @prize.levels
     end

     def getTreasuresGained
       @prize.treasures
     end
     
     def getCombattLevel
       @combatLevel
     end
     
     def getLevelChangeAgainstCultistPlayer
       return @levelChangeAgainstCultistPlayer + @combatLevel
     end
     
     def to_s
      "Name = #{@name}  CombatLevel = #{@combatLevel}"
     end

  end
end
